Modules/Core/FiniteDifference/include/itkDenseFiniteDifferenceImageFilter.hxx

```diff
@@ -41,8 +41,9 @@ DenseFiniteDifferenceImageFilter<TInputImage, TOutputImage>::CopyInputToOutput()
   // Check if we are doing in-place filtering
   if (this->GetInPlace() && this->CanRunInPlace())
   {
-    typename TInputImage::Pointer tempPtr = output.GetPointer();
-    if (tempPtr && tempPtr->GetPixelContainer() == input->GetPixelContainer())
+    const void * const inputPixelContainer = input->GetPixelContainer();
+    const auto * const tempPtr = output.GetPointer();
```

Comment:
How about?:
```c++
const typename TInputImage::PixelContainer * const inputPixelContainer = input->GetPixelContainer();
const TOutputImage * const tempPtr = output.GetPointer();
```

Reply:
`inputPixelContainer` would be typed pointer, and therefore the compile might complain about the comparison.

Reply:
Right, got it, the comparison would then be in trouble.  My religion says to use explicit types whenever possible, in this case to catch when return types in the first two lines change unexpectedly, so I might go with the following third line.  But you don't have to join my religion... please do as makes the most sense to you.
```c++
const typename TInputImage::PixelContainer * const inputPixelContainer = input->GetPixelContainer();
const TOutputImage * const tempPtr = output.GetPointer();
if (tempPtr != nullptr && (void *) tempPtr->GetPixelContainer() == (void *) inputPixelContainer)
```

Reply:
I'm fine with either `typename TInputImage::PixelContainer` (from @Leengit) or `auto` (as I originally proposed). But please please please no C-style casts. If possible, I strongly prefer no casts at all.

Reply:
Resolving following Niels' last comment. Thanks @N-Dekker and @Leengit for the discussion, insight and proposals.